Add metube extension

dotvhs opened the pull request:
## Description

This extension communicates with [MeTube](https://github.com/alexta69/metube) and allows for adding links to download quicker. It implements two commands, one that sends clipboard to MeTube and one that allows adding through a form if person needs to change Quality or Format of media being downloaded.

## Screencast

https://github.com/user-attachments/assets/5dcf4299-3f1c-45c0-9b55-d93357250e18

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer commented on extensions/metube/src/add-to-metube.tsx:
```diff
@@ -0,0 +1,79 @@
+  quality: string;
+  format: string;
+};
+
+interface Preferences {
+  metubeUrl: string;
+  quality: string;
+  format: string;
+}
```
```suggestion
```
This is already a part of `raycast-env.d.ts`

pernielsentikaer commented on extensions/metube/src/add-to-metube.tsx:
```diff
@@ -0,0 +1,79 @@
+  const format = preferences.format;
+
+  async function handleSubmit(values: Values) {
+    if (!values.url) {
```
You can use [useForm](https://developers.raycast.com/utilities/react-hooks/useform) which has validation, errors and handles states

dotvhs replied:
Oh! I see, thank you, I was not aware of this, looks interesting, I will implement this for sure.

pernielsentikaer commented on extensions/metube/src/clipboard-to-metube.ts:
```diff
@@ -0,0 +1,18 @@
+
+export default async function main() {
+  const clip = await Clipboard.readText();
+  if (clip) {
```
Here you can also validate if it's even a link (and maybe if it's a youtube?)

dotvhs replied:
I could validate if it follows a generic link pattern, sure. However, MeTube works through yt-dlp which supports lots and lots of websites and not just YT, so I don't think limiting it to just YT would be helpful in this case but I still appreciate the suggestion.

pernielsentikaer commented on extensions/metube/src/helper.ts:
```diff
@@ -0,0 +1,40 @@
+  msg?: string;
+}
+
+interface Preferences {
+  metubeUrl: string;
+  quality: string;
+  format: string;
+}
```
```suggestion
```
This is already a part of `raycast-env.d.ts`

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

dotvhs commented:
I finished requested changes. I added verification, removed the unnecessary parts of code and I'm checking if content is a link.

Hopefully I did things correct this time, thank you for your time.

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
